### ISSUE TITLE
Connecting the output of a relay to a node before the input of the relay is connected doesn't work

### ISSUE THREAD

**Issue by ykgamma** [Frooxius/NeosPublic#790]
When a relay is spawned in, the input and output both have the dummy type.

Expected Result: If I wire the output of a relay to the input of a node with a given data type, the relay connects and updates to have the data type of the input node.

Actual Result: If I wire the output of a relay to the input of a node with a given data type, the connection isn't made and the relay is left as dummy type.

Required workaround: I have to connect the input of the relay to the datatype I want, disconnect the input, and then connect the output of the relay to the other node.

**Comment by shiftyscales** [comment]
There is currently no way to overload a node from its output. This has been a requested change prior as it would make it considerably easier to use nodes such as the dynamic impulse receiver with value. To make this change would require some architectural changes to how overloading works, but could be considered in the future alongside other Logix tweaks.

**Comment by Frooxius** [comment]
Yeah, this needs a mechanism for overloading from outputs first, but that one is a fair bit tricky, as output is a lot more ambiguous and it gets weirder when propagating the swapped overload through he chain. It'll be added at some point, but it needs to be carefully designed.
